demosplan/DemosPlanCoreBundle/DoctrineMigrations/2024/02/Version20240220142305.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of the package demosplan.
+ *
+ * (c) 2010-present DEMOS plan GmbH, for more information see the license file.
+ *
+ * All rights reserved
+ */
+
+namespace Application\Migrations;
+
+use Doctrine\DBAL\Exception;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20240220142305 extends AbstractMigration
+{
+    private const BASEMAP_GIS_URL = 'https://sgx.geodatenzentrum.de/wms_basemapde%';
+    private const BASEMAP_GIS_NAME = 'basemap';
+
+    public function getDescription(): string
+    {
+        return 'refs T17533: Update _g_print setting for gislayer (which use basemap) of blueprints.
+        - Gislayer of blueprints having this criteria are updated:
+
+        a) default masterblueprint
+        b) blueprints created by users
+        c) masterblueprints set by the user (which replace the default masterblueprint)';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->abortIfNotMysql();
+        // Update gis layers where its procedure is a master template
+        $this->addSql(
+            'UPDATE _gis as g
+        INNER JOIN _procedure AS p ON g._p_id = p._p_id
+        SET g._g_print = 1
+        WHERE g._g_url like :basemapGisUrl
+        AND g._g_name = :basemapGisName
+        AND (p._p_id IN (SELECT _procedure FROM customer) OR p.master_template = 1 OR p._p_master = 1)',
```

Comment:
No blocker - just a bit redundant.
SELECT _procedure FROM customer
all of those have to be procedures where
 _p_master is 1.
Customers can only hold a relation to their default-customer-blueprint.

Reply:
Hey @MoritzMandler
- it is true. I can remove the customer part as it is redundant.
- yes it should be applicable for all projects

Reply:
@MoritzMandler I updated the sql based on Andreas feedback (I also updated the description in ticket and pr)